### DISCUSSION TITLE
Event bus not defined error

### DISCUSSION THREAD

**Discussion by alecoud** [nylo-core/nylo#187] [Q&A] [answered]
I use Auth.authenticate(data: user) to authorize login and hte below method to logout the app:

await lockRelease('logout_app', perform: () async {
                await controller.logout();
                await Auth.logout();
                await NyStorage.deleteAll(andFromBackpack: true);
                routeToInitial();
              });

Then if I try login again, it get this error:
Event bus not defined. Please ensure that your project has called nylo.addEventBus() in one of your providers.

I have to hot restart the app to be able to login again, anyone could help to analyze the reason? Thanks!

**Comment by agordn52** [comment]
> I use Auth.authenticate(data: user) to authorize login and hte below method to logout the app:
> 
> 
> 
> await lockRelease('logout_app', perform: () async {
> 
>                 await controller.logout();
> 
>                 await Auth.logout();
> 
>                 await NyStorage.deleteAll(andFromBackpack: true);
> 
>                 routeToInitial();
> 
>               });
> 
> 
> 
> Then if I try login again, it get this error:
> 
> Event bus not defined. Please ensure that your project has called nylo.addEventBus() in one of your providers.
> 
> 
> 
> I have to hot restart the app to be able to login again, anyone could help to analyze the reason? Thanks!

Hi @alecoud, which version of nylo_framework are you using? I'll take a look today

**Reply by alecoud** [reply]
Hi @agordn52 , I am using the newest version 6.3.6, it looks it's because if set andFromBackpack: true in NyStorage.deleteAll(), it deletes everything including Event bus which required by Nylo framework.

**Reply by agordn52** [reply, marked as answer]
Thanks for reporting this, I've fixed it in `nylo_framework: ^6.3.0`
You can update your pubspec.yaml 👍

**Reply by alecoud** [reply]
This update fixed the problem, many thanks!
